### ISSUE TITLE
[Windows 11] Opening the file picker may crash the host.

### ISSUE THREAD

**Issue by ShadowShok** [lsp-plugins/lsp-plugins#425]
Any time i click load, save, or the import box in any plugin, the plugin may crash, it may not crash at the first attempt, but clicking load and save multiple times, or browsing files with the plugin file picker may cause the host to crash after opening a directory.

I haven't found a consistent way to replicate the crash, sometimes it may not crash at all, I've found the crash to be more common with the Impulse Response/Reverb plugins.

Plugins built with MSYS2 UCRT64, Make 4.4.1, PHP 8.3.10, gcc 14.1.0.
make configuration: `FEATURES='lv2 vst2 vst3 clap ui doc'`

I've found versions 1.2.17 and 1.2.16 to be affected, i haven't tested any other version.
Looks like it does not depend on the type of plugin type or the host.
I've tried VST on equalizer apo, VST3 on reaper, LV2 on reaper and carla, and CLAP on reaper.

Maybe adding a make option to use the system file picker could fix the issue.

**Comment by sadko4u** [comment]
Have you tried to build plugins with `DEBUG=1` option and run the host in gdb?
At least getting stack trace could be useful.

**Comment by ShadowShok** [comment]
I ran Light Host 1.2.1 inside gdb, i was able to get the crash.
```[New Thread 2644.0x24f4]
[New Thread 2644.0x3420]
[New Thread 2644.0x1b1c]
[New Thread 2644.0x3640]
[New Thread 2644.0x3668]
[New Thread 2644.0x1e1c]
[New Thread 2644.0xc60]
[New Thread 2644.0x9d4]
[Thread 2644.0x9d4 exited with code 0]
[New Thread 2644.0x2c30]
warning: Attempting to load VST: D:\Users\Public\source\lsp-plugins-debug-1.2.17\INSTALL\lib\vst\lsp-plugins.vst\impulse-responses-stereo.dll
warning: Creating VST instance: impulse-responses-stereo
[New Thread 2644.0x1118]
warning: Initialising VST: impulse-responses-stereo (V1.0.2.4)
[New Thread 2644.0x23ec]
warning: Opening VST UI: LSP Impulse Responses Stereo [VST]
[New Thread 2644.0x3234]
[New Thread 2644.0x1614]
[New Thread 2644.0x33e8]
[Thread 2644.0x33e8 exited with code 0]
[New Thread 2644.0x2718]
[New Thread 2644.0x3c90]
[New Thread 2644.0xfe8]
[New Thread 2644.0x38b0]
[New Thread 2644.0xb70]
[New Thread 2644.0x1754]
[New Thread 2644.0x1390]
[New Thread 2644.0x1664]
[New Thread 2644.0x2ffc]
[New Thread 2644.0x2ed4]
[New Thread 2644.0x37b4]
[Thread 2644.0x37b4 exited with code 0]

Thread 1 received signal SIGSEGV, Segmentation fault.
0x00007fff8db4ba58 in lsp::io::Path::canonicalize (this=this@entry=0x14ef60)
    at main/io/Path.cpp:1254
warning: 1254   main/io/Path.cpp: No such file or directory
(gdb) continue
Continuing.

Thread 1 received signal SIGSEGV, Segmentation fault.
0x00007fff8db4ba58 in lsp::io::Path::canonicalize (this=this@entry=0x14ef60)
    at main/io/Path.cpp:1254
1254    in main/io/Path.cpp
(gdb) continue
Continuing.
[Thread 2644.0x1b1c exited with code 3221225477]
[Thread 2644.0x1e1c exited with code 3221225477]
[Thread 2644.0x3668 exited with code 3221225477]
[Thread 2644.0xc60 exited with code 3221225477]
[Thread 2644.0x2c30 exited with code 3221225477]
[Thread 2644.0x1118 exited with code 3221225477]
[Thread 2644.0x1614 exited with code 3221225477]
[Thread 2644.0x3640 exited with code 3221225477]
[Thread 2644.0xfe8 exited with code 3221225477]
[Thread 2644.0x2718 exited with code 3221225477]
[Thread 2644.0x38b0 exited with code 3221225477]
[Thread 2644.0x1390 exited with code 3221225477]
[Thread 2644.0x1754 exited with code 3221225477]
[Thread 2644.0xb70 exited with code 3221225477]
[Thread 2644.0x2ffc exited with code 3221225477]
[Thread 2644.0x1664 exited with code 3221225477]
[Thread 2644.0x24f4 exited with code 3221225477]
[Thread 2644.0x3234 exited with code 3221225477]
[Thread 2644.0x23ec exited with code 3221225477]
[Thread 2644.0x36d8 exited with code 3221225477]
[Thread 2644.0x3c90 exited with code 3221225477]
[Thread 2644.0x3420 exited with code 3221225477]

Program terminated with signal SIGTRAP, Trace/breakpoint trap.
The program no longer exists.
```

**Comment by sadko4u** [comment]
OK, much better. Now we know that function `lsp::io::Path::canonicalize` fails. I think we can add some debug message to see the path that causes the problem. Are you available to build from the `devel` branch of the Git repository?

**Comment by sadko4u** [comment]
It also could be nice if you show the stack trace of the proble by issuing the `bt` command after the host crashes.

**Comment by sadko4u** [comment]
Was able to reproduce the problem on my virtual machine.

**Comment by sadko4u** [comment]
Just committed changes to [lsp-runtime-lib](https://github.com/lsp-plugins/lsp-runtime-lib/commit/62ff5f925d2f35b323ab5a4fece5c93d7053e398).

Please try to build from the `devel` branch.

**Comment by ShadowShok** [comment]
I built the `devel` repo and I'm no longer experiencing the issue.

**Comment by sadko4u** [comment]
Nice.

**Comment by sadko4u** [comment]
Fixed in 1.2.18.
